### PR TITLE
bug 1772607: retry image mirror failures in periodic mirroring job

### PR DESCRIPTION
**Pull request by bparees**
adding retries based on https://bugzilla.redhat.com/show_bug.cgi?id=1772607#c5

### CODE REVIEW

**Comment by bparees** [comment]
/assign @smarterclayton 

as the keeper of this job i'll let you decide if you're ok paving over this problem or not.

File: ci-operator/jobs/infra-image-mirroring.yaml [hunk 1]
```diff
@@ -15,11 +15,15 @@ periodics:
         set -o errexit
         failures=0
         for mapping in /etc/imagemirror/mapping_knative*; do
-          echo "Running: oc image mirror -f=$mapping --skip-multiple-scopes"
-          if ! oc image mirror -f="$mapping" --skip-multiple-scopes; then
-            echo "ERROR: Failed to mirror images from $mapping"
-            failures=$((failures+1))
-          fi
+          for count in {1..3}; do
```
Comment (smarterclayton) [inline ci-operator/jobs/infra-image-mirroring.yaml:18]: For right now, set --continue-on-error as well (assuming you have a new enough oc).  That makes sure we actually mirror everything without an error.

Comment (bparees) [reply ci-operator/jobs/infra-image-mirroring.yaml:18]: this job was using quay.io/openshift/origin-cli:4.3 which didn't have it. i've updated the job to use quay.io/openshift/origin-cli:4.5 which does, and added the flag.

Comment (stevekuznetsov) [reply ci-operator/jobs/infra-image-mirroring.yaml:18]: this is using `quay.io/openshift/origin-cli:4.3` -- not sure why, might as well use `registry.svc.ci.openshift.org/ocp/4.5:cli`

Comment (bparees) [reply ci-operator/jobs/infra-image-mirroring.yaml:18]: updated to your suggestion, @stevekuznetsov

**Comment by bparees** [comment]
@stevekuznetsov @smarterclayton addressed your comments

**Comment by bparees** [comment]
Looks like there already was retry logic in the openshift mirroring job and we're still seeing failures after 3 attempts:
https://storage.googleapis.com/origin-ci-test/logs/periodic-image-mirroring-openshift/6917/build-log.txt

```
sha256:e0c102530035dcfe17f1729e41717f31e7d1168892a66037d5c3cba2ae3566ef quay.io/openshift/origin-ovn-kubernetes:latest
error: unable to push manifest to quay.io/openshift/origin-cluster-authentication-operator:latest: error parsing HTTP 400 response body: invalid character '<' looking for beginning of value: "<html>\r\n<head><title>400 Bad Request</title></head>\r\n<body bgcolor=\"white\">\r\n<center><h1>400 Bad Request</h1></center>\r\n<hr><center>nginx/1.12.1</center>\r\n</body>\r\n</html>\r\n"
info: Mirroring completed in 14.88s (0B/s)
I0503 16:12:37.881431     183 workqueue.go:54] worker 5 stopping
I0503 16:12:37.881457     183 workqueue.go:54] worker 0 stopping
I0503 16:12:37.881472     183 workqueue.go:54] worker 3 stopping
I0503 16:12:37.881503     183 workqueue.go:54] worker 0 stopping
I0503 16:12:37.881489     183 workqueue.go:54] worker 1 stopping
I0503 16:12:37.881508     183 workqueue.go:54] worker 2 stopping
I0503 16:12:37.881517     183 workqueue.go:54] worker 1 stopping
I0503 16:12:37.881511     183 workqueue.go:60] work queue exiting
I0503 16:12:37.881494     183 workqueue.go:60] work queue exiting
F0503 16:12:37.881485     183 helpers.go:114] error: one or more errors occurred while uploading images
I0503 16:12:37.881490     183 workqueue.go:54] worker 4 stopping
I0503 16:12:37.881488     183 workqueue.go:54] worker 3 stopping
I0503 16:12:37.881542     183 workqueue.go:54] worker 2 stopping
ERROR: Failed to mirror images from /etc/imagemirror/mapping_origin_4_5 after 3 attempts
```

(it failed on a different image in each of the 3 tries)

https://github.com/openshift/release/blob/4e40b9911217953fbeb3ad06f57d92cfa6543c0c/ci-operator/jobs/infra-image-mirroring.yaml#L70-L79

This PR affects a completely different job (I wasn't paying attention).  either we should copy the existing logic to all the other mirroring jobs, or just close this PR.

/hold

**Pull request closed without merging**